datastream-server-restli/src/test/java/com/linkedin/datastream/server/TestCoordinator.java

```diff
@@ -3230,6 +3230,7 @@ void testOnSessionExpired(boolean handleNewSession) throws DatastreamException,
       Assert.assertTrue(t != null && t.isAlive());
       Assert.assertTrue(PollUtils.poll(() -> instance1.getIsLeader().getAsBoolean(), 100, 30000));
     }
+    instance1._coordinatorEventThreadExiting = false;
```

Comment:
interesting that we have to set this explicitly 🤔

Reply:
The test do zkExpiry and then handleNewSession and then call stop. So during zkExpiry, we go from false -> true, and during stop if `_coordinatorEventThreadExiting ` stays false, then wait does not occur and creates deadlock.